File: project/src/main/java/ro/ubb/project/core/model/Paper.java
package ro.ubb.project.core.model;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Paper implements Serializable {
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Paper paper = (Paper) o;
        return pid == paper.pid;
    }

    @Override
    public int hashCode() {
        return Objects.hash(pid);
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int pid;
    private int aid;
    private int session;
    private String title;
    private String accepted;
    @Column(name="abstracturl")
    private String abstractUrl;
    @Column(name="contenturl")
    private String contentUrl;
}

File: project/src/main/webapp/web-client/src/app/model/paper.ts
export class Paper {
  pid: number;
  authorId: number;
  session: number;
  title: string;
  topic: string;
  accepted: string;
  abstractUrl: string;
  contentUrl: string;


  constructor(pid: number, aid: number, session: number, title: string, topic: string, accepted: string, abstractUrl: string, contentUrl: string) {
    this.pid = pid;
    this.authorId = aid;
    this.session = session;
    this.title = title;
    this.topic = topic;
    this.accepted = accepted;
    this.abstractUrl = abstractUrl;
    this.contentUrl = contentUrl;
  }

}
